index.js

```diff
@@ -25,13 +25,31 @@ var UNICODE_CHARS = {
     '\u2029': '\\u2029'
 };
 
+// We can‘t just instanceof Date since dates are already converted to strings
+// because of native Date.prototype.JSON (which use toISOString)
+var DATE_LENGTH = new Date().toISOString().length
+function isDate(d) {
+  try {
+    return (
+      // testing length first to avoid new Date() for every string
+      d.length === DATE_LENGTH &&
```

Comment:
I understand, I'm just saying that you should probably apply a regexp to prevent other arbitrary strings with the same length to be treated as dates just because they have the exact same length. Otherwise an error will throw.